challenges/2021-04-20-potential-palindrome/solutions/parhaml.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+# Potential Palindrome
+#
+# Given a string, determine whether any permutation of it is a palindrome.
+# A palindrome is any string that can be read the same both forwards and
+# backwards, such as "kayak".
+#
+# ## Business Rules/Errata
+#
+# - You only need to determine whether it is possible to make a palindrome,
+# you do not need to return an example of the palindrome.
+# - Any input that is not a string should be treated as a string, if possible.
+# - An empty string is not a palindrome.
+#
+# ## Examples
+#
+# ```bash
+# canMakePalindrome("carrace");  // True
+# ```
+#
+# The string `carrace` can be rearranged to make `racecar`, a palindrome.
+
+require 'test/unit/assertions'
+include Test::Unit::Assertions
+
+def can_make_palindrome(str)
+  str = str.to_s
+
+  return 'An empty string cannot make a palindrome' if str.empty?
+
+  result_hsh = word_count(str)
+  odd_values = result_hsh.reject { |_, v| v.even? }.count
+
+  odd_values <= 1 ? "'#{str}' can make a palindrome" : "'#{str}' can not make a palindrome"
```

Comment:
Does this mean Ruby is expression-based? (i.e., you don't need to explicitly 'return' from functions) Do things like: `var_name = condition ? value_if_true : value_if_false` work?

Reply:
Correct. Ruby implicitly returns the last thing run